### PR TITLE
PN-256: Non-consented patients are considered as matches for incoming MME requests

### PR DESCRIPTION
**Pull request by allasm**
@sashaandjic please test (I will explain how on Slack)

### CODE REVIEW

File: similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java [hunk 1]
```diff
@@ -111,13 +115,19 @@ public void initialize() throws InitializationException
     @Override
     public List<PatientSimilarityView> findSimilarPatients(Patient referencePatient)
     {
-        return find(referencePatient, false);
+        return findSimilarPatients(referencePatient, null);
```
Comment (veronikaslc) [inline similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java]: why not simply `return find(referencePatient, null, false);`

File: similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java [hunk 2]
```diff
@@ -111,13 +115,19 @@ public void initialize() throws InitializationException
...
+    }
+
+    @Override
+    public List<PatientSimilarityView> findSimilarPatients(Patient referencePatient, String requiredConsentId)
```
Comment (veronikaslc) [inline similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java]: Why not the list of all possible (in future) consents? We have only one just for now.

File: similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java [hunk 3]
```diff
@@ -144,6 +154,10 @@ public long countSimilarPatients(Patient referencePatient)
                 // Leftover patient in the index, should be removed
                 continue;
             }
+            // check consents, if required
+            if (requiredConsentId != null && !consentManager.hasConsent(matchPatient, requiredConsentId)) {
```
Comment (veronikaslc) [inline similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java]: either `check consent` or it should a be a list of consents

File: similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java (hunk 1 above)
Comment (allasm) [reply similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java]: more modular, this method is a shortcut for `findSimilarPatients` for the case when we have no consents, so w edo exactly that here. The fact that that method in turn calls something else is its internal implementation which we dont care about here, maybe in the future that will change, but this way we will only have to change it in one place (ther,e not both there and here)

File: similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java (hunk 2 above)
Comment (allasm) [reply similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java]: I thought about this, just no sure what is a good convenient way to do this (just alist may work, but is ugly on the calling side, when you have to create a list of one item all the time). We can upgrade to include a list later when we have a better idea about a use case

Comment (sdumitriu) [reply similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java]: varargs is easy.

**Pull request merged**